system/jlib/jtrace.hpp

```diff
@@ -67,6 +80,9 @@ interface ISpan : extends IInterface
     virtual void toString(StringBuffer & out) const = 0;
     virtual void getLogPrefix(StringBuffer & out) const = 0;
     virtual bool isRecording() const = 0;   // Is it worth adding any events/attributes to this span?
+    virtual void recordException(IException * e, bool spanFailed = true) = 0;
+    virtual void recordError(const SpanError error = SpanError(), bool spanFailed = true) = 0;
```

Comment:
c++ coding:  SpanError should be a reference parameter - otherwise it will be cloned each time it is passed (depending on the compiler).